api/mutations/channel/archiveChannel.js

```diff
@@ -1,36 +1,19 @@
 // @flow
 import type { GraphQLContext } from '../../';
 import UserError from '../../utils/UserError';
-import { getUserPermissionsInChannel } from '../../models/usersChannels';
-import { getUserPermissionsInCommunity } from '../../models/usersCommunities';
 import { getChannelById, archiveChannel } from '../../models/channel';
+import { userCanManageChannel } from './utils';
 
 export default async (
   _: any,
   { input: { channelId } }: { input: { channelId: string } },
-  { user }: GraphQLContext
+  { user, loaders }: GraphQLContext
 ) => {
-  const currentUser = user;
-
-  // user must be authed to delete a channel
-  if (!currentUser) {
-    return new UserError(
-      'You must be signed in to make changes to this channel.'
-    );
+  if (await !userCanManageChannel(user.id, channelId)) {
```

Comment:
It'd be better if these lived in the GraphQL context so you

a) wouldn' have to pass the user ID manually every time
b) **you could use the loaders (properly)**

```JS
const resolver = (root, args, { permissions }) => {
  if (!(await permissions.canManageChannel(channelId)) { /* ... */ }
}
```

This would be implemented something like this:

```JS
const createPermissionsMethods = (userId: string, loaders) => ({
  canManageChannel: (channelId: string) => { return loaders.userChannelPermissions.load(userId, channelId) }
})

graphqlExpress(req => {
  const loaders = createLoaders();
  return {
    context: {
     loaders,
     permissions: createPermissionsMethods(req.user && req.user.id, loaders),
    }
  }
})
```

Alternatively we could also overload these onto the `user` key, which might be even clearer than `permissions`:

```JS
const resolver = (root, args, { user }) => {
  if (!(await user.canManageChannel(channelId)) { /* ... */ }
}
```